### PR TITLE
fix(alias): properly initialize custom resolvers

### LINKED ISSUE #203: Plugin Alias v3 throws error
- Rollup Plugin Name: Alias
- Rollup Plugin Version: 3.0.1
- Rollup Version: 1.31.0
- Operating System (or Browser): macOS 10.15.3
- Node Version: 13.8.0

### How Do We Reproduce?

[https://repl.it/repls/WatchfulUnsightlyStruct](https://repl.it/repls/WatchfulUnsightlyStruct)

### Expected Behavior
Expect to compile as per v2.2.0

### Actual Behavior
Throws mysterious error `(plugin alias) TypeError: Cannot read property 'input' of undefined`

Thanks!

### PR DESCRIPTION
**Pull request by lukastaegert**
## Rollup Plugin Name: `alias`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:
Resolves #203 

### Description
This will make sure that for all custom resolvers
- the `buildStart` hooks are run if present during `buildStart`
- the `resolveId` hook is run with correct `this` context

### CODE REVIEW

File: packages/alias/src/index.ts [hunk 1]
```diff
@@ -61,6 +79,17 @@ export default function alias(options: RollupAliasOptions = {}): Plugin {
...
+    buildStart(inputOptions) {
+      return Promise.all(
+        [...entries, options].map(
+          ({ customResolver }) =>
+            customResolver &&
+            typeof customResolver === 'object' &&
+            typeof customResolver.buildStart === 'function' &&
+            customResolver.buildStart.call(this, inputOptions)
+        )
+      ).then(() => {});
```
Comment (pkit) [inline packages/alias/src/index.ts:91]: Shouldn't something be caught here?

Comment (lukastaegert) [reply packages/alias/src/index.ts:91]: What do you expect to catch? If something fails, then Rollup's `buildStart` will pick it up. The empty `then` is just to satisfy the expected empty return type of `buildStart`.

Comment (lukastaegert) [reply packages/alias/src/index.ts:91]: We could also do a type cast here of course, but I do not think this is performance critical.

Comment (pkit) [reply packages/alias/src/index.ts]: > just to satisfy the expected empty return type

Ah! Sorry

**Pull request merged**